app/services/solr_mapper.rb

```diff
@@ -123,4 +172,45 @@ def struct_fields
   def related_identifiers_field
     metadata['related_identifiers']&.pluck('related_identifier')
   end
+
+  def publisher_id_field
+    metadata.dig('publisher', 'publisher_identifier')
+  end
+
+  def publisher_field
+    metadata.dig('publisher', 'name')
+  end
+
+  def subjects_field
+    metadata['subjects']&.pluck('subject')
+  end
+
+  # A range should be specified as [date]/[date] where date can be in
+  # multiple formats. See dataworks_schema.yml for details.
+  def date_range_years(date_value)
+    return [] unless date_value.split('/').length == 2
+
+    years = date_value.split('/').map do |date_section|
+      date_year(date_section)
+    end
+    # Create range of years including the beginning and end years provided
+    Array(years[0]..years[1])
+  end
+
+  # Retrieve just the year from a particular date string
+  def date_year(date_value)
+    # An allowable schema date format is just the year e.g. '2024'
+    return date_value.to_i if date_value.length == 4
+
+    # Date.parse will work with both 'YYYY-MM-DD' and 'YYYY- MM-DDThh:mm:ssTZD'
+    Date.parse(date_value).year
+  end
+
+  # Retrieve affiliation name array given either creator or contributor field
+  def affiliation_names_for_role(role)
+    metadata[role]&.map do |role_entity|
```

Comment:
Can you use `flat_map` instead of adding `flatten`?

Also, this is another case where `Array(metadata[role])` will allow a much cleaner statement (e.g., not needing all of the safe navigation).

Reply:
Yes.